src/main/java/geocode/GeoName.java

```diff
@@ -92,15 +95,13 @@ protected double axisSquaredDistance(GeoName other, int axis) {
         return distance * distance;
     }
 
+    /*
+     * Distance formulas taken from https://www.movable-type.co.uk/scripts/latlong-vectors.html
+     */
     @Override
-    protected double haversineDistance(GeoName other, double radius) {
-        double dLat = toRadians(other.latitude - this.latitude);
-        double dLon = toRadians(other.longitude - this.longitude);
-        double lat1 = toRadians(this.latitude);
-        double lat2 = toRadians(other.latitude);
-
-        double a = pow(sin(dLat / 2),2) + pow(sin(dLon / 2),2) * cos(lat1) * cos(lat2);
-        return 2 * asin(sqrt(a)) * radius;
+    protected double distance(GeoName other, double radius) {
+        // distance = atan2(cross product, dot product) * radius
+        return atan2(cross(other), dot(other)) * radius;
```

Comment:
This looks good, as do the cross and dot product functions 👍 

Took me a second to figure out where the magnitude calculation came from but it was right too it looks like